src/aiogram_dialog/widgets/kbd/select.py

```diff
@@ -181,7 +181,7 @@ async def _process_click(
             await self.on_item_click.process_event(
                 callback, select, manager, self.type_factory(item_id),
             )
-        await self._on_click(callback, select, manager, str(item_id))
```

Comment:
So, would it be correct to return type casting in `Select` and remove it in `StatefulSelect`? Also make a condition if `item_id` is the member of enum, then pass it as `self.item_id_getter(item_id)` instead of just casting to str.